specs/schemas/schema-schema.ipldsch

```diff
@@ -699,19 +723,73 @@ type EnumRepresentation union {
 ## in the DSL) which will be stored here in the representation block. Missing
 ## entries in this map will use the default.
 ##
-type EnumRepresentation_String {EnumValue:String}
+type EnumRepresentation_String {EnumMember:String}
 
 ## EnumRepresentation_Int describes the way an enum is represented as an int
 ## in the data model. A mapping of names to ints is required to perform the
 ## conversion from int to enum value. In the DSL, int values _must_ be provided
-## for each EnumValue (with `Foo ("100")`, those are stored here.
+## for each EnumMember (with `Foo ("100")`, those are stored here.
+##
+type EnumRepresentation_Int {EnumMember:Int}
+
+## TypeUnit describes a type which contains no data at all (other than that fact of its existence).
+## (If this seems strange, consider that the cardinality of TypeBool is 2;
+## the cardinality of TypeUnit is simply 1.)
 ##
-type EnumRepresentation_Int {EnumValue:Int}
+type TypeUnit struct {
+	representation UnitRepresentation
+}
+
+## UnionRepresentation is an enum for describing how a TypeUnit should be represented in the data model.
+##
+## Unit types are commonly seen represented in several ways.
+## A null token is a common one.
+## A true token is sometimes seen (especially, when people encode "sets" in json:
+## often this will be seen as a map where the values are keys and the map values are 'true').
+## Also, an empty map can be a useful unit value;
+## an empty map accurately communicates a lack of data.
+## (The emptymap strategy can be a particularly interesting choice if you want to
+## have a schema that is evolvable in the future to start using a struct or map
+## in the same place as the unit type currently stands, while having older documents
+## continue to be parsable by the evolved schema.)
+##
+## Unlike many of the other representation information types seen in the schema-schema,
+## this one is just an enum, rather than being a union.
+## That's because there's no possibility of every needing to annotate more customization
+## onto values in the unit type... because there are no possible values in the unit type.
+##
+## Note that there is no discernible logical difference between
+## `type Foo struct {}` and `type Foo unit representation emptymap`;
+## only that the latter can be said to be a more explicit description of intent.
+## Both will result in identical representations, and both have identical cardinality (which is 1).
+##
+type UnitRepresentation enum {
+	| Null ("null")
+	| True ("true")
+	| False ("false")
+	| Emptymap ("emptymap")
+}
+
+## TypeAny describes a type which can contain data of any kind.
+## It's essentially an escape valve; it says "we don't really know what lies beyond here".
+## The type-level data model kind of an "any" type can be anything;
+## it depends on what the inhabitant value is.
+## The representation-level kind can similarly be anything;
+## it will match whatever the type-level kind is.
```

Comment:
```suggestion
## The representation-level kind can similarly be any data valid according to the IPLD data model;
## it will match whatever the type-level kind is.
```

goes without saying in ipld-prime, but elsewhere, like JS, `any` doesn't let you give me a `Date` or `RegExp` object

Reply:
hrm, does the term 'kind' not make it explicit enough?

Okay, yeah, I guess I buy that this could be clearer.  The sentence right above probably needs a patch too then, though, as it's saying a very similar thing.